### PR TITLE
Moving function_schema to aten/core

### PR DESCRIPTION
**Pull request by bwasti**
Summary: Couple more moves to unblock IR unification

Differential Revision: D9819906

### CODE REVIEW

File: aten/src/ATen/core/function_schema.h [hunk 1]
```diff
@@ -0,0 +1,117 @@
+#pragma once
+#include "ATen/ATen.h"
```
Comment (ezyang) [inline aten/src/ATen/core/function_schema.h]: You can't do this, `ATen/ATen.h` pulls in a lot of other stuff that's not core, and if you actually use those symbols you'll get a linker error in envs that only link against source files in core. Try deleting it and see what you actually need.

File: aten/src/ATen/core/interned_strings.h [hunk 2]
```diff
@@ -0,0 +1,260 @@
+#pragma once
+#include <vector>
+#include <cstdint>
+#include <string>
+#include <unordered_map>
+#include <algorithm>
+
+#include "torch/csrc/jit/generated/aten_interned_strings.h"
```
Comment (ezyang) [inline aten/src/ATen/core/interned_strings.h]: We talked about this: the plan is to remove this `#include` from `interned_strings.h` entirely

**Review by ezyang — CHANGES_REQUESTED** [review]
aten_interned_strings.h needs to be handled

**Comment by fmassa** [comment]
@bwasti could you look into addressing @ezyang comments?

**Pull request closed without merging**